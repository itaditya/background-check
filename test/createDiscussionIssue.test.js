const createDiscussionIssue = require('../lib/github-api/createDiscussionIssue')

process.env.APP_ORG_NAME = 'probot-background-check'

test('createDiscussionIssue is working', async () => {
  const github = {
    issues: { create: jest.fn() }
  }

  await createDiscussionIssue(github, {
    owner: 'probot',
    username: 'itaditya',
    toxicComments: [{
      link: 'link1',
      text: 'text1',
      toxicScore: 0.8
    }, {
      link: 'link2',
      text: 'text2',
      toxicScore: 0.9
    }]
  })

  expect(github.issues.create.mock.calls).toMatchSnapshot()
})
