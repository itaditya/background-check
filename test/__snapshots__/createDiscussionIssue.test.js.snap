// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDiscussionIssue is working 1`] = `
Array [
  Array [
    Object {
      "body": "Some comments from @itaditya have been found to be toxic. Review the following comments and discuss whether or not you would like to allow this contributor to participate in your community.


   1. link2 has a toxicity rating of **0.9** :

       > text2
       


   1. link1 has a toxicity rating of **0.8** :

       > text1
       
",
      "owner": "probot-background-check",
      "repo": "probot-discussions",
      "title": "itaditya-discussion",
    },
  ],
]
`;
